### PR TITLE
[#4310] Added headers to DApps list

### LINKED ISSUE #4310: Add category headers to DApp screen
### User Story

As a user, I want to know what types of DApps are available by category, so I have better context into what they do.

### Description

*Type*: Feature

*Summary*: 
Use text headers to break the DApp list into categories:

- Exchanges: Airswap, Bancor, ERC dEX, Kyber, Oasis Direct
- Marketplaces: CryptoCribs, Ethlance, OpenSea, Name Bazaar
- Fun & Games: CryptoKitties, CryptoFighters, CryptoPunks, Etherbots
- Social Networks: Cent, Peepeth
- Utilities: Hexel

#### Expected behavior

"Open DApp" screen presents a list with category titles. 

#### Actual behavior

"Open DApp" screen contains an alphabetical list of DApps.

### PR DESCRIPTION
**Pull request by jeluard**
fixes #4310

### Summary:

Use list with headers to display DApps.

### Steps to test:
- Open Status
- Open a DApp
- Browse DApps in a list with headers

status: ready

### CODE REVIEW

**Comment by denis-sharypin** [comment]
Screenshot bitte

**Comment by jeluard** [comment]
@denis-sharypin I can see you are fluent in german now!

list

**Comment by denis-sharypin** [comment]
Can you make the space between tittle and previous list 24px
screenshot 2018-05-22 10 59 01

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/WC3eaP for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/WC3eaP">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/ui/screens/group/events.cljs [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+            [re-frame.core :as re-frame]
+            [status-im.utils.js-resources :as js-res]
+            [status-im.utils.handlers :as handlers]
+            [status-im.ui.screens.group.navigation])
+  (:require-macros [status-im.utils.slurp :refer [slurp]]))
+
+;;;; COFX
+
+(re-frame/reg-cofx
+ :get-default-contacts
```
Comment (flexsurfer) [inline src/status_im/ui/screens/group/events.cljs]: why did you add this cofx?

File: src/status_im/ui/screens/group/events.cljs [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+ (fn [coeffects _]
+   (assoc coeffects
+          :default-contacts js-res/default-contacts)))
+
+(re-frame/reg-cofx
+ :get-default-dapps
+ (fn [coeffects _]
+   (assoc coeffects
+          :default-dapps (reader/read-string (slurp "./src/status_im/default-dapps.edn")))))
```
Comment (flexsurfer) [inline src/status_im/ui/screens/group/events.cljs]: what the reason to have edn instead clj?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/fKxrfe for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/fKxrfe">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/ui/screens/group/events.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/group/events.cljs]: Just followed the pattern used for contacts. Should it be different?

Comment (flexsurfer) [reply src/status_im/ui/screens/group/events.cljs]: i mean `:get-default-contacts` looks like it's by mistake here, we have it in contacts ns

File: src/status_im/ui/screens/group/events.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/ui/screens/group/events.cljs]: Will change to clj

File: src/status_im/ui/screens/group/events.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/group/events.cljs]: Hmm merge probably.

**Comment by jeluard** [comment]
@denis-sharypin I am using our default component. This change would then apply to others list with sections (like tx history).

**Comment by denis-sharypin** [comment]
@jeluard yes please!

**Comment by jeluard** [comment]
@denis-sharypin Currently margin-top is 11 on android and 10 on iOS. So 24 everywhere? What about margin-bottom?

**Comment by denis-sharypin** [comment]
@jeluard let's make `margin-top`  16px. Current `margin-bottom` is alright.

**Comment by jeluard** [comment]
@denis-sharypin @flexsurfer Done!

File: src/status_im/ui/screens/group/events.cljs [hunk 3]
```diff
@@ -1,6 +1,9 @@
 (ns status-im.ui.screens.group.events
   (:require [status-im.utils.handlers :as handlers]
-            [status-im.ui.screens.group.navigation]))
+            [status-im.ui.screens.group.navigation])
+  (:require-macros [status-im.utils.slurp :refer [slurp]]))
+
+;;;; COFX
```
Comment (flexsurfer) [inline src/status_im/ui/screens/group/events.cljs]: maybe revert this file?

File: src/status_im/ui/screens/contacts/subs.cljs [hunk 4]
```diff
@@ -42,12 +42,7 @@
            (remove :dapp? contacts)))
 
 (reg-sub :all-dapp-with-url-contacts
-         :<- [:all-added-contacts]
-         :<- [:get-current-account]
-         (fn [[contacts {:keys [dev-mode?]}]]
-           (filter #(and (:dapp? %) (:dapp-url %) (or dev-mode?
-                                                      (not= "simple-dapp" (:whisper-identity %))))
-                   contacts)))
+         :contacts/dapps)
```
Comment (flexsurfer) [inline src/status_im/ui/screens/contacts/subs.cljs]: we still need to filter "simple-dapp" here

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/db.cljs [hunk 5]
```diff
@@ -139,6 +139,7 @@
 (spec/def ::db (allowed-keys
                 :opt
                 [:contacts/contacts
+                 :contacts/dapps
```
Comment (flexsurfer) [inline src/status_im/ui/screens/db.cljs:148]: do you think dapps should be in contacts?

File: resources/default_contacts.json [hunk 6]
```diff
@@ -18,142 +18,6 @@
...
-        "photo-path": "contacts://cryptocribs"
-    },
-    "cryptofighters": {
-        "name" : {"en" : "CryptoFighters"},
-        "description": "Collect train and fight digital fighters.",
-        "dapp-url": {"en" : "https://cryptofighters.io"},
-        "dapp?": true,
-        "photo-path": "contacts://cryptofighters"
-    },
     "simple-dapp": {
```
Comment (flexsurfer) [inline resources/default_contacts.json]: why you didn't move  "simple-dapp"?

File: src/status_im/ui/screens/db.cljs (hunk 5 above)
Comment (jeluard) [reply src/status_im/ui/screens/db.cljs:148]: In the apps they are still contacts and some code does the same assumption. Maybe something that will change in the next redesign? ;)

File: src/status_im/ui/screens/group/events.cljs (hunk 3 above)
Comment (jeluard) [reply src/status_im/ui/screens/group/events.cljs]: ok

File: resources/default_contacts.json (hunk 6 above)
Comment (jeluard) [reply resources/default_contacts.json]: simple-dapp is not part of the new section based design. Maybe it should be removed from default completely?

Comment (flexsurfer) [reply resources/default_contacts.json]: @rachelhamlin can we add `For developers` sections? or something, simple-dapp is important for testing and developing purposes, it should be visible only in dev mode

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/2p3BEH for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/2p3BEH">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/tYS8Mm for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/tYS8Mm">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/bcnAEo for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/bcnAEo">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/pxyvup for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/pxyvup">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/ui/components/contact/contact.cljs [hunk 7]
```diff
@@ -20,7 +19,7 @@
                          :number-of-lines 1}
                         (when dapp? {:accessibility-label :dapp-name})
                         props)
-      (if (pos? (count (:name contact)))
```
Comment (goranjovic) [inline src/status_im/ui/components/contact/contact.cljs:23]: `(seq (:name contact))`

**Review by goranjovic — APPROVED** [review]

**Comment by jeluard** [comment]
@flexsurfer done

**Review by flexsurfer — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/2rrNkv for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/2rrNkv">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/Gq8Rbh for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/Gq8Rbh">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/Rt7DqY for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/Rt7DqY">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/dZi3AA for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/dZi3AA">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/Sx4XNs for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/Sx4XNs">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/hvfBJt for branch PR-4375

<blockquote><div><strong><a href="https://i.diawi.com/hvfBJt">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
